.github/workflows/build-and-test.yml

```diff
@@ -27,9 +27,22 @@ jobs:
           key: v1-${{ runner.os }}-gradle-${{ github.ref }}-${{ github.sha }}
           restore-keys: v1-${{ runner.os }}-gradle-${{ github.ref }}
 
+      - name: Cache SonarCloud packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+
       - name: Gradle build service
         run: ./gradlew --build-cache :service:build -x test
 
+      - name: SonarQube scan
+        run: ./gradlew --build-cache :service:sonarqube
```

Comment:
Running sonarqube this way won't pick up coverage information from your build. If you want coverage data as part of the sonar scan, you must run it after running `test`.

The good news is, once you've done this you can remove `codecov` from your workflow.

For an example, see https://github.com/DataBiosphere/terra-data-catalog/blob/main/.github/workflows/build-and-test.yml#L121

Reply:
It looks like you also need to configure XML coverage output from `jacoco` if you aren't doing that already.

https://github.com/DataBiosphere/terra-data-catalog/blob/main/buildSrc/src/main/groovy/bio.terra.catalog.java-common-conventions.gradle#L97

Reply:
Good ideas! Perhaps I can update the recommendations. From AppSec's point of view, we don't require coverage reporting but it is nice in the sonarqube status, and if it is better than codecov then hey why not.